### PR TITLE
Unum password etc

### PR DESCRIPTION
**Pull request by roblevintennis**
**Description:**
- [Asana Task](https://app.asana.com/0/1136962714401929/1144809619356373/f)
- Password field for Unum create page

**Screenshots:**

Invalid characters missing:

image

Invalid empty:

image

Valid:

image

### CODE REVIEW

File: src/components/Inputs/TextInput/TextInput.js [hunk 1]
```diff
@@ -108,12 +111,18 @@ PrivateTextInput.PUBLIC_PROPS = {
   validator: PropTypes.func,
   onBlur: PropTypes.func,
   onFocus: PropTypes.func,
+  restrictIllegal: PropTypes.bool,
```
Comment (roblevintennis) [inline src/components/Inputs/TextInput/TextInput.js]: We need to be able to allow special characters for password. Reality is we urlencode at the KOA side anyway so this should be fine I think.

File: src/validators/validatePassword.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const passwordRegex = /^(?=.*\d)(?=.*[A-Z])(?=.*[a-z])(?=.*[!#$%^&*])[a-zA-Z0-9!#$%^&*]{8,100}$/
```
Comment (roblevintennis) [inline src/validators/validatePassword.js]: Regex explanation:
`(?=` is a positive lookahead, and we're looking for:
- digits
- a-zA-Z (at least one upper and one lower for example)
- special characters
- 8 min - 100 max characters

Above is parity with what I saw in the legacy application, and seems to be a common approach so I just implemented it here.

Comment (ethantran) [reply src/validators/validatePassword.js]: This is not allowing `@`. A legacy thing? If we are not allow that, then the default settings for most password managers like LastPass and 1Password could be rejected.

Comment (kenning) [reply src/validators/validatePassword.js]: me and rob are looking around at this. but from lingke's email on 'bias for action:' 

> Sometimes the data is hard to gather, or the data itself becomes philosophically debated. In these cases, if the cost is small, and the action is easily reversible, just do it and see how it goes. Remember, mistakes are okay - especially if we can easily reverse the decision.

this is just a regex and can be updated really quickly, i'm fine with this for now and if someone complains or wants to look further into it we can investigate then.

File: src/validators/validatePassword.test.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  })
+
+  test('missing uppercase letter', () => {
+    expect(validatePassword('abcabc!23').length > 0).toBe(true)
+  })
+
+  test('missing lowercase letter', () => {
+    expect(validatePassword('ABCAABC!23').length > 0).toBe(true)
+  })
+})
```
Comment (kenning) [inline src/validators/validatePassword.test.js]: Nice tests

File: src/validators/validatePassword.js (hunk 2 above)
Comment (roblevintennis) [reply src/validators/validatePassword.js]: Yeah, it's fine:

>     Make special characters more permissive with @-_ characters now allowed.

It's hard to find great data on this but:
https://discussions.agilebits.com/discussion/71544/only-certain-symbols-are-allowed-when-creating-secure-passwords

> I was just trying to change my password on the Capital One site https://www.capitalonecardservice.com/ and I ran into the following message:
> 
> "Passwords must be between 8-32 characters using letters, numbers and at least one special character @./|$*&-_"
> 

Also, looks like 1Password is kinda dynamic and has some sort of database per website??

> It is possible that they previously allowed that and now do not, or that our list of disallowed characters for that site is outdated. If the latter that should hopefully be addressed on our end in an update.

So my "bias for action" is to just add those 3 characters.

As @hilem mentioned, it's easy to update :)

Comment (roblevintennis) [reply src/validators/validatePassword.js]: I should also say I appreciate the perspective you both brought to this. I elected to add in `@` and `-` in the spirit of @ethantran comment, and give up trying on the `_` in the spirit of @kenning 🤣 🤣 🤣 🤣 🤣 🤣

**Pull request merged**